auacm/app/static/js/controllers/EditCompetitionController.js

```diff
@@ -0,0 +1,223 @@
+app.controller('CreateCompetitionController', ['$scope', '$http', '$location',
+        '$routeParams', function($scope, $http, $location, $routeParams) {
+    $scope.compProblems = [];
+    var competition;
+
+    if ($routeParams.cid !== undefined) {
```

Comment:
I vote that if we find that a library redefines `undefined`, we don't use it, but we can cross that bridge when we come to it.